### DISCUSSION TITLE
How to run?

### DISCUSSION THREAD

**Discussion by viperwasp** [ggml-org/llama.cpp#1136] [Q&A]
I used to use Llama.cpp 2 weeks ago. I knew how to run it back when it has a file named "Main" and I used a batfile which included the following. 

title llama.cpp
:start
main -i --interactive-first -r "### Human:" --temp 0 -c 2048 -n -1 --ignore-eos --repeat_penalty 1.2 --instruct -m ggml-model-q4_1.bin
pause
goto start

This does not work now. I imagine it's outdated. 
I put my models in the models folder. Now I need to know how to actually use this program. 
Thanks. 

I learned more.
I read into it. I have to like download visual studios, use Cmake etc. Quantize models? The old llama.cpp I downloaded just worked. Oh well I will slowly learn to how operate this nightmare of a software. lol I also have to download weights like that?

**Comment by BetaDoggo** [comment]
Prebuilt windows binaries are still available from the [releases section](https://github.com/ggerganov/llama.cpp/releases/tag/master-ec9cdb6). If you place your model in the same folder as the executables your current script should still work.

You may have to reconvert and quantize the model to get it to work with the newest version or use the [convert.py](https://github.com/ggerganov/llama.cpp/blob/master/convert.py) script from this repo to convert the current model to the new format.

**Reply by viperwasp** [reply]
> Prebuilt windows binaries are still available from the [releases section](https://github.com/ggerganov/llama.cpp/releases/tag/master-ec9cdb6). If you place your model in the same folder as the executables your current script should still work.
> 
> You may have to reconvert and quantize the model to get it to work with the newest version or use the [convert.py](https://github.com/ggerganov/llama.cpp/blob/master/convert.py) script from this repo to convert the current model to the new format.

Thank you BetaDoggo!
I got this in my command window when I tried to run my bat file. 



**F:\AI2\llama-master-cc9cee8-bin-win-avx-x64 - CPU New April>title llama.cpp**

**F:\AI2\llama-master-cc9cee8-bin-win-avx-x64 - CPU New April>main -i --interactive-first -r "### Human:" --temp 0 -c 2048 -n -1 --ignore-eos --repeat_penalty 1.2 --instruct -m ggml-model-q4_1.bin -t 18
'main' is not recognized as an internal or external command,
operable program or batch file.**

**F:\AI2\llama-master-cc9cee8-bin-win-avx-x64 - CPU New April>pause
Press any key to continue . . .**



Pressing any key loops it? 
I had the model in the same folder as the bat file. What am I doing wrong? Or will this just not work.

**Reply by BetaDoggo** [reply]
It looks like you need to replace `main` in your script with with `main.exe`.

**Reply by viperwasp** [reply]
Thanks but that does not work. There is also no longer a file called main in the new llama.cpp directory that was from the old llama. So I don't even know what to do.

**Reply by BetaDoggo** [reply]
The prebuilt versions from the releases section do come with a main.exe. Make sure that you are using the [current release](https://github.com/ggerganov/llama.cpp/releases/download/master-8a0f867/llama-master-8a0f867-bin-win-avx2-x64.zip) and that your script and model are in the same folder as the executables.

**Reply by viperwasp** [reply]
Thanks I somehow obtained an entirely different download that did not contain main.
